src/x509.rs

```diff
@@ -268,6 +270,12 @@ fn read_four_char<S: Source>(source: &mut S) -> Result<u32, S::Err> {
 #[derive(Clone, Copy, Debug)]
 pub struct ValidationError;
 
+impl fmt::Display for ValidationError{
```

Comment:
`#[derive(Fail)]` and `[fail(display="validation error")]` instead?

Reply:
done